File: Freedom/我的数据库/MyDatabaseViewController.m
//  MyDatabaseViewController.m
//  Freedom
//  Created by Super on 16/8/18.
//  Copyright © 2016年 Super. All rights reserved.
//
#import "MyDatabaseViewController.h"
#import "MyDatabaseEditViewController.h"
#import "User.h"
@interface DatabaseCollectionViewCell : BaseCollectionViewCell
@end
@implementation DatabaseCollectionViewCell
-(void)initUI{
    self.icon = [[UIImageView alloc]initWithFrame:CGRectMake(10,0, APPW/5-20,40)];
    self.title = [[UILabel alloc]initWithFrame:CGRectMake(0,YH(self.icon), APPW/5-12, 20)];
    self.title.font = fontnomal;self.title.textAlignment = NSTextAlignmentCenter;
    [self addSubviews:self.title,self.icon,nil];
}
-(void)setCollectionDataWithDic:(NSDictionary *)dict{
    self.title.text = [dict valueForKey:@"title"];
    self.icon.image = [UIImage imageNamed:[dict valueForKey:@"icon"]];
}
@end
@interface MyDatabaseViewController()<UICollectionViewDelegateFlowLayout>{}
@end
@implementation MyDatabaseViewController
- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"我的数据库";
    UIBarButtonItem *left = [[UIBarButtonItem alloc]initWithTitle:@"设置" style:UIBarButtonItemStylePlain actionBlick:^{}];
    UIBarButtonItem *right = [[UIBarButtonItem alloc]initWithImage:[UIImage imageNamed:Padd] style:UIBarButtonItemStyleDone actionBlick:^{}];
    self.navigationItem.leftBarButtonItem  = left;
    self.navigationItem.rightBarButtonItem = right;
    BaseCollectionViewLayout *layout = [BaseCollectionViewLayout sharedFlowlayoutWithCellSize:CGSizeMake((APPW-50)/4, 60) groupInset:UIEdgeInsetsMake(0, 10, 0, 10) itemSpace:10 linespace:10];
    
    layout.headerReferenceSize = CGSizeMake(APPW, 30);layout.footerReferenceSize = CGSizeZero;
    self.collectionView = [[BaseCollectionView alloc]initWithFrame:CGRectMake(0, 0, APPW, APPH-110) collectionViewLayout:layout];
    self.collectionView.dataArray = [NSMutableArray arrayWithArray:[User getControllerData]];
    [self fillTheCollectionViewDataWithCanMove:NO sectionN:1 itemN:15 itemName:@"DatabaseCollectionViewCell"];
    self.collectionView.dataSource = self;
    self.collectionView.delegate = self;
    self.collectionView.frame = self.view.bounds;
    [self.view addSubview:self.collectionView];
}
-(void)viewWillAppear:(BOOL)animated{
    self.tabBarController.tabBar.hidden = YES;
    self.edgesForExtendedLayout = UIRectEdgeLeft | UIRectEdgeRight|UIRectEdgeBottom;
}
- (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPath:(NSIndexPath *)indexPath {
    NSString *log = [NSString stringWithFormat:@"你选择的是%zd，%zd", indexPath.section, indexPath.row];
    [SVProgressHUD showSuccessWithStatus:log];DLog(@"%@",log);
    [self pushController:[MyDatabaseEditViewController class] withInfo:self.collectionView.dataArray[indexPath.row] withTitle:@"数据库编辑详情"];
}
@end

File: Freedom/我的数据库/MyDatabaseViewController.swift
//
//  MyDatabaseViewController.swift
//  Freedom
import UIKit
import XExtension
import XCarryOn
class MyDatabaseViewController: MyDatabaseBaseViewController{
    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = UIColor.darkGray
        let starsView = Circular3DStarView(frame: CGRect(x: 0, y: 100, width: APPW, height: APPW))
        var stars: [UIView] = []
        let items = ["微信","支付宝","酷狗","爱奇艺","抖音","淘宝","新浪微博","大众点评","今日头条","微能量","微信阅读","个性特色自由主义"]
        for i in 0..<100 {
            let index = Int(arc4random() % 12)
            let text = items[index]
            let item = UILabel(frame: CGRect(x: 0, y: 0, width: 100, height: 30), font: UIFont.systemFont(ofSize: 10), color: UIColor.white, text: text)
            item.textAlignment = .center
            item.isTapEffectEnabled = false
            let pointView = UIButton(frame: CGRect(x: 45, y: 20, width: 10, height: 10))
            pointView.layer.cornerRadius = 5
            pointView.backgroundColor = UIColor.random
            pointView.tag = i
            item.addSubview(pointView)
            stars.append(item)
        }
        starsView.stars = stars
        view.addSubview(starsView)
    }
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        self.tabBarController?.tabBar.isHidden = true
        self.edgesForExtendedLayout = .all
    }

    @objc func touchedItem(_ sender: UIButton) {
        let log = "你选择的是\(sender.tag)"
        noticeInfo(log, autoClear: true, autoClearTime: 3)
        push(MyDatabaseEditViewController(), info: "", title: "数据库编辑详情")
    }
}
